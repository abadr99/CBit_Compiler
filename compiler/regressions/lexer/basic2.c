// RUN: cbit %s --keep_lexer=true 
// RUN: diff basic2.lex basic2.expected 
// RUN: rm -rf basic2.lex

int add(int a, int b) {
    return a+b;
}

int main() {
    int a = add(2,3);
    return 0;
}