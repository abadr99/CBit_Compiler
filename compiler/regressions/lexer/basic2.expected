Token-Value                                                                     Token-Type
int                                                                             KeyWord
add                                                                             ID
(                                                                               Operator
int                                                                             KeyWord
a                                                                               ID
,                                                                               Symbol
int                                                                             KeyWord
b                                                                               ID
)                                                                               Symbol
{                                                                               Operator
return                                                                          KeyWord
a                                                                               ID
+                                                                               Operator
b                                                                               ID
;                                                                               Symbol
}                                                                               Symbol
int                                                                             KeyWord
main                                                                            ID
(                                                                               Operator
)                                                                               Symbol
{                                                                               Operator
int                                                                             KeyWord
a                                                                               ID
=                                                                               Operator
add                                                                             ID
(                                                                               Operator
2                                                                               Number
,                                                                               Symbol
3                                                                               Number
)                                                                               Symbol
;                                                                               Symbol
return                                                                          KeyWord
0                                                                               Number
;                                                                               Symbol
}                                                                               Symbol
